cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaZkTopicConfigProvider.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.config;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import com.linkedin.kafka.cruisecontrol.KafkaCruiseControlUtils;
+import com.linkedin.kafka.cruisecontrol.config.constants.ExecutorConfig;
+import kafka.server.ConfigType;
+import kafka.zk.AdminZkClient;
+import kafka.zk.KafkaZkClient;
+import scala.collection.JavaConversions;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Properties;
+
+
+/**
+ * The Kafka topic config provider implementation based on files. The format of the file is JSON, listing properties:
+ * <pre>
+ *   {
+ *     "min.insync.replicas": 1,
+ *     "an.example.cluster.config": false
+ *   }
+ * </pre>
+ *
+ */
+public class KafkaZkTopicConfigProvider implements TopicConfigProvider {
```

Comment:
I have added deprecated to the ZK provider and switched the names around. 

I have made the Kafka Admin provider the default? Is that how you want to proceed or do you want to have a release or two with the ZK provider as the primary still and then switch over?

Reply:
It would indeed be useful to have at least one release before we change the default value. So this patch would add support and a followup patch would make it the default.